### ISSUE TITLE
Disabling Piped proxy doesn't actually disable connecting to one?

### ISSUE THREAD

**Issue by code-consensus** [libre-tube/LibreTube#6366]
### Steps to reproduce

1. Go to Settings / Instance
2. Toggle "Disable Piped proxy" (i.e. choose to _not_ use a Piped instance)

### Expected behavior

The video plays (_directly_ from YouTube's servers)

### Actual behavior

The video doesn't load, and gives an error "Got error: Sign in to confirm that you're not a bot"
(this is the same error affecting the Piped instances being blocked by YouTube, but here no Piped instance should be used)

### LibreTube version

v0.24.0

### Android version

Android 14

### Other details

There is a well-known issue affecting Piped instances getting a sign-in error due to being blocked by YouTube (detailed in [Piped issue 3658](https://github.com/TeamPiped/Piped/issues/3658)), but the bug being reported here is for _**not**_ using a Piped proxy.

There have been many issues created in the last couple of months reporting a "confirm you're not a bot" error, but the resolution posted to each of those closed issues is to change the Piped instance in the settings (e.g. issue #6116) -- the problem I am reporting here is that this solution doesn't apply if one is **_not_** _trying to use any Piped instance at all_.

I have tried side-by-side, same phone and same phone IP address, one-after-the-other, loading the same video with the NewPipe app (which connects directly to YouTube's servers), a browser (which also connects directly to YouTube's servers) and LibreTube with "Disable Piped proxy" set -- NewPipe and the browser will play the video, whereas LibreTube gives the "confirm you're not a bot" error.

Given that NewPipe loads the video, I wonder whether the issue is that perhaps that the "Disable Piped proxy" toggle in LibreTube is somehow _not_ working as expected, such that the app is still trying to somehow connect via a proxy? That may explain why you still get the bot error, despite not trying to use a (blocked) Piped instance.

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

**Comment by Bnyro** [comment]
Disabling the Piped proxy does exactly what it is expected to. It no longer proxies video streams and images through Piped.

Video information such as the video stream source to load the actual video from are still fetched from Piped's API, what's giving the error message described above.

**Comment by executed** [comment]
Do I understand it correctly that even when all settings force app to avoid Piped proxy it still might hit the errors because Piped API server is not functioning smoothly?

What is recommended workaround in this case - switching the Piped Proxy until one works?
If so - were there any discussions to automatically choose Piped API server that is up and running specifically to just query video stream source to load the actual video from?

**Comment by Bnyro** [comment]
I've been working on support for using the app completely independent of Piped, see https://github.com/libre-tube/LibreTube/pull/7151 for instance.

These changes will be part of the next release.
